### PR TITLE
Reimplement pretty-printing through Kiama

### LINKED ISSUE #12: Implement pretty-printing for types and separate it from Type.toString
As for terms and as we learned in PTS, we need both the default toString and pretty-printing as string representations for different use cases, instead of stretching a single representation for both needs.

In a perfect world, it would be cool to reuse some Scala port of a Haskell pretty-printing library (what we use is often inconvenient because it uses side effects, and lacks quite a few features). Last time I didn't find anything (but should look closer), not sure I'll be hacking that soon.

### PR DESCRIPTION
**Pull request by yfcai**
Fix #12.

* Use Kiama's pretty-printing library.

* Add pretty printer for types.

* Use Kiama's pretty printer for terms.

Please review for style adherence, sufficient testing and obvious bugs.

### CODE REVIEW

**Comment by Blaisorblade** [comment]
Cai, thanks a lot for this! I'll take a look and take over.

File: src/main/scala/ilc/feature/sums/Pretty.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package ilc
+package feature
+package sums
+
+/**
+ * Pretty printing for first-class functions.
```
Comment (Blaisorblade) [inline src/main/scala/ilc/feature/sums/Pretty.scala]: Typo.

File: src/test/scala/ilc/feature/let/BetaReductionSuite.scala [hunk 2]
```diff
@@ -83,8 +83,8 @@ extends FlatSpec
   val duplicating = lambda(Var("f", NatType =>: NatType), Var("expensive", NatType)) {
     case Seq(arg, expensive) => PlusNat ! (arg ! expensive) ! (arg ! expensive)
   } ! u ! (PlusNat ! 42 ! 84)
-  pretty(duplicating)
-  pretty(normalize(duplicating))
+  info("\n" + pretty(duplicating))
+  info("\n" + pretty(normalize(duplicating)))
```
Comment (Blaisorblade) [inline src/test/scala/ilc/feature/let/BetaReductionSuite.scala]: Don't we have some other method for this?

File: src/main/scala/ilc/feature/sums/Pretty.scala (hunk 1 above)
Comment (yfcai) [inline src/main/scala/ilc/feature/sums/Pretty.scala]: Ouch.

File: src/test/scala/ilc/feature/let/BetaReductionSuite.scala (hunk 2 above)
Comment (yfcai) [inline src/test/scala/ilc/feature/let/BetaReductionSuite.scala]: [1]: https://github.com/inc-lc/ilc-scala/blob/topic/pretty-by-kiama/src/test/scala/ilc/feature/functions/PrettySuite.scala#L16

The method is in [`feature.functions.PrettySuite`][1]. The call to `pretty` here looks provisional, and there's some commented-out code block below. I assumed this test is in the middle of development by @Blaisorblade.

Comment (Blaisorblade) [inline src/test/scala/ilc/feature/let/BetaReductionSuite.scala]: Indeed, the old code looks indeed broken (probably coming from a worksheet).
I was thinking of [show()](https://github.com/inc-lc/ilc-scala/blob/a8f616b7f44a9cb0092b95af314ec40047b69d13/src/test/scala/ilc/feature/let/Instantiations.scala#L8), which again was used for worksheets so it just returns the string.

Comment (Blaisorblade) [inline src/test/scala/ilc/feature/let/BetaReductionSuite.scala]: Anyway, quite some mess to clean up.

Comment (Blaisorblade) [inline src/test/scala/ilc/feature/let/BetaReductionSuite.scala]: Improved (but not fixed) in bada36e5b7ab5f48e62828f418ed81b36ab559e9.

**Comment by Blaisorblade** [comment]
I plan to merge this as soon as tests pass, since I've addressed all concerns critical to me, so I believe this is already, overall, an improvement.

**Pull request merged**